### ISSUE TITLE
Create CLI interface to interact with the library

### ISSUE THREAD

**Issue by david-alvarez-rosa** [Waifod/quant_finance_models#33]
Create CLI interface to interact with the library.

Something similar to 

```sh
./qfm --asset <my_asset> --model black_scholes 
```

**Comment by david-alvarez-rosa** [comment]
I was tentatively planning to use Boost library `program_options` for generating the CLI interface.  However, I'm not able to link, hence I'm blocked in this path until https://github.com/boostorg/program_options/issues/125 is solved.

Other alternatives are using GNU `getopt` or Google `gflags`, but I'd prefer to use Boost.

**Comment by waifod** [comment]
I agree on using `Boost` since we are already including it for other things. We can wait, I think.

**Comment by david-alvarez-rosa** [comment]
@Waifod—solved.

**Comment by david-alvarez-rosa** [comment]
Started on #91

**Comment by david-alvarez-rosa** [comment]
@Waifod, I'd need a bit of brainstorm of use cases we would like to support. Thoughts?

**Comment by waifod** [comment]
Shall we discuss them in person?
